modules/gapi/test/common/gapi_video_tests_inl.hpp

```diff
@@ -131,7 +131,51 @@ TEST_P(BackgroundSubtractorTest, AccuracyTest)
     // Allowing 1% difference of all pixels between G-API and reference OpenCV results
     testBackgroundSubtractorStreaming(gapiBackSub, pOCVBackSub, 1, 1, learningRate, testNumFrames);
 }
+
+TEST_P(KalmanFilterTest, AccuracyTest)
+{
+    cv::gapi::video::KalmanParams kp(dDim, mDim, cDim, depth);
+
+    //measurement vector
+    cv::Mat measure_vec(mDim, 1, depth);
+    cv::randu(measure_vec, Scalar::all(-1), Scalar::all(1));
+
+    //control vector
+    cv::Mat ctrl_vec = Mat::zeros(cDim > 0 ? cDim : dDim, 1, depth);
+
+    if (cDim > 0)
+        cv::randu(ctrl_vec, Scalar::all(-1), Scalar::all(1));
+
+    // G-API Kalman state
+    cv::Mat gapiKState(dDim, 1, depth);
+    // OCV Kalman state
+    cv::Mat ocvKState(dDim, 1, depth);
+
+    // G-API code
+    cv::GMat m, ctrl;
+    cv::GOpaque<bool> have_m;
+    cv::GMat out = cv::gapi::KalmanFilter(m, have_m, ctrl, kp);
+    cv::GComputation comp(cv::GIn(m, have_m, ctrl), cv::GOut(out));
+
+    comp.apply(cv::gin(measure_vec, haveMeasure, ctrl_vec), cv::gout(gapiKState));
+
+    // OpenCV code
+    cv::KalmanFilter ocvKalman(dDim, mDim, cDim, depth);
+
+    ocvKState = cDim > 0 ? ocvKalman.predict(ctrl_vec) : ocvKalman.predict();
+
+    if (haveMeasure)
+        ocvKState = ocvKalman.correct(measure_vec);
+
+    // Comparison //////////////////////////////////////////////////////////////
+    {
+        double diff = 0;
+        vector<int> idx;
+        EXPECT_TRUE(cmpEps(gapiKState, ocvKState, &diff, 1.0, &idx, false) >= 0);
+    }
+}
```

Comment:
Please could you add the test with the several iterations of the algorithm, it will clarify how to use our kernel.

Reply:
I meant something like that:
```
    //measurement vector
    cv::Mat measure_vec(mDim, 1, depth);

    //control vector
    cv::Mat ctrl_vec = Mat::zeros(cDim > 0 ? cDim : dDim, 1, depth);

    // G-API Kalman state
    cv::Mat gapiKState(dDim, 1, depth);
    // OCV Kalman state
    cv::Mat ocvKState(dDim, 1, depth);

    // G-API graph initialization
    cv::GMat m, ctrl;
    cv::GOpaque<bool> have_m;
    cv::GMat out = cv::gapi::KalmanFilter(m, have_m, ctrl, kp);
    cv::GComputation comp(cv::GIn(m, have_m, ctrl), cv::GOut(out));

    // OpenCV reference KalmanFilter initialization
    cv::KalmanFilter ocvKalman(dDim, mDim, cDim, depth);

   int N = 5; // number of iterations (measurements to correct) - can be a test parameter

   cv::RNG& rng = cv::theRNG();

    for (int i = 0; i < N; i++)
   {   
         bool haveMeasure = rng(2u); // returns 0 or 1

         cv::randu(measure_vec, Scalar::all(-1), Scalar::all(1));
         if (cDim > 0)
            cv::randu(ctrl_vec, Scalar::all(-1), Scalar::all(1));

        // G-API code
        comp.apply(cv::gin(measure_vec, haveMeasure, ctrl_vec), cv::gout(gapiKState));

        // OpenCV code
       ocvKState = cDim > 0 ? ocvKalman.predict(ctrl_vec) : ocvKalman.predict();
        if (haveMeasure)
            ocvKState = ocvKalman.correct(measure_vec);
```

Reply:
Ok. Applied.